geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/AccumuloIndexAdapter.scala

```diff
@@ -0,0 +1,452 @@
+
+      if (index.name == IdIndex.name) {
+        // enable the row functor as the feature ID is stored in the Row ID
+        tableOps.setProperty(table, Property.TABLE_BLOOM_KEY_FUNCTOR.getKey, classOf[RowFunctor].getCanonicalName)
+        tableOps.setProperty(table, Property.TABLE_BLOOM_ENABLED.getKey, "true")
+      }
+    }
+  }
+
+  // noinspection ScalaDeprecation
+  override def deleteTables(tables: Seq[String]): Unit = {
+    tables.par.foreach { table =>
+      if (tableOps.exists(table)) {
+        if (ds.connector.isInstanceOf[org.apache.accumulo.core.client.mock.MockConnector]) {
+          // we need to synchronize deleting of tables in mock accumulo as it's not thread safe
+          ds.connector.synchronized(tableOps.delete(table))
+        } else {
+          tableOps.delete(table)
+        }
+      }
+    }
+  }
+
+  override def clearTables(tables: Seq[String], prefix: Option[Array[Byte]]): Unit = {
+    val auths = ds.auths // get the auths once up front
+    tables.par.foreach { table =>
+      if (tableOps.exists(table)) {
+        val config = GeoMesaBatchWriterConfig().setMaxWriteThreads(ds.config.writeThreads)
+        WithClose(ds.connector.createBatchDeleter(table, auths, ds.config.queryThreads, config)) { deleter =>
+          val range = prefix.map(p => Range.prefix(new Text(p))).getOrElse(new Range())
+          deleter.setRanges(Collections.singletonList(range))
+          deleter.delete()
+        }
+      }
+    }
+  }
+
+  override def createQueryPlan(strategy: QueryStrategy): AccumuloQueryPlan = {
+    import org.locationtech.geomesa.index.conf.QueryHints.RichHints
+
+    if (strategy.ranges.isEmpty) { EmptyPlan(strategy.filter) } else {
+      val QueryStrategy(filter, byteRanges, _, _, ecql, hints, _) = strategy
+      val index = filter.index
+      // index api defines empty start/end for open-ended range - in accumulo, it's indicated with null
+      // index api defines start row inclusive, end row exclusive
+      val ranges = byteRanges.map {
+        case BoundedByteRange(start, end) =>
+            val startKey = if (start.length == 0) { null } else { new Key(new Text(start)) }
+            val endKey = if (end.length == 0) { null } else { new Key(new Text(end)) }
+            new Range(startKey, true, endKey, false)
+
+        case SingleRowByteRange(row) =>
+          new Range(new Text(row))
+      }
+      val numThreads = if (index.name == IdIndex.name) { ds.config.recordThreads } else { ds.config.queryThreads }
+      val tables = index.getTablesForQuery(filter.filter)
+      val (colFamily, schema) = {
+        val (cf, s) = groups.group(index.sft, hints.getTransformDefinition, ecql)
+        (Some(new Text(AccumuloIndexAdapter.mapColumnFamily(index)(cf))), s)
+      }
+
+      index match {
+        case i: AccumuloJoinIndex =>
+          i.createQueryPlan(filter, tables, ranges, colFamily, schema, ecql, hints, numThreads)
```

Comment:
Does the Join Index support Arrow/Bin/Heatmaps/Stats?

Reply:
yep!